src/main/scala/sangria/macros/derive/DeriveObjectTypeMacro.scala

```diff
@@ -251,6 +256,32 @@ class DeriveObjectTypeMacro(context: blackbox.Context) extends {
     def unknownMember(pos: Position, name: String) =
       pos → s"Unknown member '$name'. Known members are: ${knownMembers map (_.name) mkString ", "}"
 
+    def getMethod(pos: Position, name: String) =
+      knownMembers.withFilter(_.name == name).map(_.method) match {
+        case method :: Nil ⇒ Right(method)
+        case Nil ⇒ Left(unknownMember(pos, name) :: Nil)
+        case _ ⇒ Left(List(
+          pos → s"Cannot configure overloaded method '$name' using `DeriveObjectSetting` due to ambiguity, use annotations instead."
+        ))
+      }
+    def getArgument(pos: Position, methodName: String, argName: String) = getMethod(pos, methodName).right.flatMap{ method ⇒
+      val knownArguments = method.paramLists.flatten
+      knownArguments.find(_.name.decodedName.toString == argName)
+        .map(Right(_))
+        .getOrElse(Left(List(
+          pos → s"Unknown argument '$argName' of method '$method'. Known arguments are: ${knownArguments.map(_.name.decodedName) mkString ", "}"
+        )))
+    }
+    def validateHasArgument(pos: Position, methodName: String, argName: String) = getArgument(pos, methodName, argName).right.map(_ ⇒ Nil).merge
+    def validateArgumentDefault(pos: Position, methodName: String, argName: String, default: Tree, argType: Type) =
+      getArgument(pos, methodName, argName).right.map{
+        case arg if argType <:< arg.typeSignature.resultType ⇒ Nil
+        case arg ⇒ List(
+          pos → s"Wrong type of default value '$default' for argument '$argName' of method '$methodName': expected '${arg.typeSignature.resultType}', but got '$argType'."
```

Comment:
I think we would need to rely on the standard set of validations for a default value type. It is based on `ArgumentType`, `ToInput` type-classes as well as some run-time validations. It still will provide the same compile-time guarantees. It's a minor thing, I will adjust it after the merge.